sdk/identity/azure-identity/azure/identity/_constants.py

```diff
@@ -48,4 +48,4 @@ class EnvironmentVariables:
     AZURE_REGIONAL_AUTHORITY_NAME = "AZURE_REGIONAL_AUTHORITY_NAME"
 
     AZURE_FEDERATED_TOKEN_FILE = "AZURE_FEDERATED_TOKEN_FILE"
-    TOKEN_EXCHANGE_VARS = (AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_FEDERATED_TOKEN_FILE)
+    TOKEN_EXCHANGE_VARS = (AZURE_AUTHORITY_HOST, AZURE_TENANT_ID, AZURE_FEDERATED_TOKEN_FILE)
```

Comment:
Is this by design?

I see we have code:

```py
        elif all(os.environ.get(var) for var in EnvironmentVariables.TOKEN_EXCHANGE_VARS):
            _LOGGER.info("%s will use token exchange", self.__class__.__name__)
            from .token_exchange import TokenExchangeCredential

            self._credential = TokenExchangeCredential(
                tenant_id=os.environ[EnvironmentVariables.AZURE_TENANT_ID],
                client_id=os.environ[EnvironmentVariables.AZURE_CLIENT_ID],
                token_file_path=os.environ[EnvironmentVariables.TOKEN_FILE_PATH],
                **kwargs
            )
```

Reply:
Yes. I've updated that code as well, did you copy it from an old revision?

The gist is, AZURE_AUTHORITY_HOST, AZURE_TENANT_ID, AZURE_FEDERATED_TOKEN_FILE are certain to have values in this environment. AZURE_CLIENT_ID may not have a value, in which case we require a value for client_id.